src/kernels/piecewisepolynomial.jl

```diff
@@ -10,13 +12,13 @@ processes are hence v times  mean-square differentiable. The kernel function is:
 where r is the Mahalanobis distance mahalanobis(x,y) with `maha` as the metric.
 
 """
-struct PiecewisePolynomialKernel{T1<:T1, T2<:Real, A<:AbstractMatrix{T2}} <: BaseKernel
-    v::T1
+struct PiecewisePolynomialKernel{V, A<:AbstractMatrix{<:Real}} <: BaseKernel
+    v::Integer
     maha::A
-    function PiecewisePolynomialKernel(v::T1, maha::AbstractMatrix{T2}) where {T1<:Integer,T2<:Real}
+    function PiecewisePolynomialKernel(v, maha::AbstractMatrix{<:Real})
```

Comment:
This would be changed to
```suggestion
    function PiecewisePolynomialKernel{V}(maha::AbstractMatrix{<:Real}) where V
```
If a user should be allowed to define a kernel with `PiecewisePolynomialKernel(v, maha)` then you should define an outer constructor
```julia
PiecewisePolynomialKernel(v::Integer, maha::AbstractMatrix{<:Real}) = PiecewisePolynomialKernel{v}(maha)
```
However, that won't be type stable.